src/nvim/api/ui.c

```diff
@@ -524,6 +525,32 @@ void nvim_ui_pum_set_bounds(uint64_t channel_id, Float width, Float height, Floa
   ui->pum_pos = true;
 }
 
+/// Send a UI event to Nvim.
+///
+/// The following events are supported:
+///
+///   - "termresponse": TUI only. The terminal sent a OSC response sequence to Nvim. The
+///                     payload is the received OSC sequence.
+///
+/// @param channel_id
+/// @param event Event name
+/// @param payload Event payload
+/// @param[out] err Error details, if any.
+void nvim_ui_event(uint64_t channel_id, String event, Object value,
```

Comment:
I'd like this to be a bit more specific. Any api call sent non-blocking is already an "event". maybe `nvim_ui_term_event()` and the discriminator then could be `osc_response` (maybe we want to distinguish more custom sequences in the future)

Reply:
Changed in 5f77b62e022ba7b7d0e0a525f95ff6cf99b5391c.